File: Python/programmers/etc/level1/두_개_뽑아서_더하기.py
def solution(numbers):
    answer = set()
    for i in range(len(numbers)-1):
        for j in range(i+1, len(numbers)):
            answer.add(numbers[i]+numbers[j])
    answer = sorted(list(answer))
    return answer


numbers = [2,1,3,4,1]
print(solution(numbers)) # [2,1,3,4,1]

numbers = [5,0,2,7]
print(solution(numbers)) # [2,5,7,9,12]

File: Java/programmers/etc/level1/두개뽑아서더하기.java
package CodeTest.Java.programmers.etc.level1;

import java.util.*;

public class 두개뽑아서더하기 {
    public List<Integer> solution(int[] numbers) {
        List<Integer> answer = new ArrayList<>();
        HashSet<Integer> set = new HashSet<>();

        int n = numbers.length;
        for (int i=0; i<n-1; i++) {
            for (int j=i+1; j<n; j++) {
                set.add(numbers[i]+numbers[j]);
            }
        }

        Iterator<Integer> iter = set.iterator();
        while (iter.hasNext()) {
            answer.add(iter.next());
        }

        Collections.sort(answer);
        return answer;
    }    
}
